src/commands/interface-manager/ParamaterParsing.ts

```diff
@@ -24,15 +24,34 @@ limitations under the License.
  * are NOT distributed, contributed, or committed under the Apache License.
  */
 
-import { Keyword, ReadItem, SuperCoolStream } from "./CommandReader";
+import { ISuperCoolStream, Keyword, ReadItem, SuperCoolStream } from "./CommandReader";
+import { PromptOptions } from "./PromptForAccept";
 import { CommandError, CommandResult } from "./Validation";
 
-export class ArgumentStream extends SuperCoolStream<ReadItem[]> {
+export interface IArgumentStream extends ISuperCoolStream<ReadItem[]> {
+    rest(): ReadItem[],
+    isPromptable(): boolean,
+    // should prompt really return a new stream?
+    prompt(paramaterDescription: ParamaterDescription): Promise<CommandResult<ReadItem>>,
+}
+
+export class ArgumentStream extends SuperCoolStream<ReadItem[]> implements IArgumentStream {
     public rest() {
         return this.source.slice(this.position);
     }
+
+    public isPromptable(): boolean {
+        return false;
+    }
+
+    prompt(paramaterDescription: ParamaterDescription): Promise<CommandResult<ReadItem>> {
+        throw new TypeError("This argument stream is NOT promptable, did you even check isPromptable().");
+    }
 }
 
+// TODO: Presentation types should be extracted to their own file.
+// FIXME: PresentationTypes should not be limited to ReadItems.
```

Comment:
Feel like this needs to be a follow up along with presentation type translators

Reply:
At the least an issue should be created saying that we need to allow the argument stream to contain things other than `ReadItem`s